ompi/mca/fcoll/static/fcoll_static_file_write_all.c

```diff
@@ -976,12 +975,12 @@ mca_fcoll_static_file_write_all (mca_io_ompio_file_t *fh,
         decoded_iov = NULL;
     }
 
-    if (my_aggregator == fh->f_rank) {
+    if (NULL != local_iov_array){
+        free(local_iov_array);
+        local_iov_array = NULL;
+    }
 
-        if (NULL != local_iov_array){
-            free(local_iov_array);
-            local_iov_array = NULL;
-        }
+    if (my_aggregator == fh->f_rank) {
```

Comment:
Reading through this function hurts ;)

Reply:
@bosilca don't worry too much about this function. The static component might be axed before the branch to 4.0